account_analytic_policy/model/account_move_line.py

```diff
@@ -0,0 +1,72 @@
+# coding: utf-8
+############################################################################
+#    Module Writen For Odoo, Open Source Management Solution
+#
+#    Copyright (c) 2010 Vauxoo - http://www.vauxoo.com
+#    All Rights Reserved.
+#    info Vauxoo (info@vauxoo.com)
+#    coded by: nhomar <nhomar@vauxoo.com>
+############################################################################
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class AccountAccount(models.Model):
+    _inherit = "account.account"
+
+    analytic_policy = fields.Selection(
+        [('optional', 'Optional'), ('always', 'Always'), ('never', 'Never')],
+        default='optional',
+        help='"Optional": , the accountant is free to put an analytic account '
+             'on an account move line;\n'
+             '"Always": the accountant will get an error message if there is '
+             'no analytic account;\n'
+             '"Never": the accountant will get an error message if an '
+             'analytic account is present.'
+    )
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.multi
+    def _is_policy_ok(self, vals):
+        account = ('account_id' in vals) and self.env[
+            'account.account'].browse(vals['account_id'])
+        analytic = ('analytic_account_id' in vals) and self.env[
+            'account.account'].browse(vals['analytic_account_id'])
```

Comment:
analytic = self.env['account.analytic.account'].browse(vals['analytic_account_id']) if ('analytic_account_id' in vals) else self. analytic_account_id

Reply:
I need to check first the key is in the dict if not the vals['analytic_account_id'] will brake.